File: testdata/eval/combined-polyglot-scope/repo/services/billing/src/refund.ts
// TypeScript side of the billing service. The fixture task
// mentions refundCharge + RefundContext; tier-2 TS extraction
// surfaces them as module-level symbols.

export interface RefundContext {
  chargeId: string;
  amount: number;
}

export function refundCharge(ctx: RefundContext): Promise<void> {
  return Promise.resolve();
}

export class RefundDispatcher {
  dispatch(ctx: RefundContext) {
    return refundCharge(ctx);
  }
}

File: testdata/eval/combined-polyglot-scope/repo/services/billing/scripts/refund.py
"""Python side of the billing service. Tier-2 (tree_sitter_python)
exposes refund_charge and RefundReport as module-level symbols."""


class RefundReport:
    """Captures refund outcomes for downstream reconciliation."""

    def __init__(self, charge_id: str):
        self.charge_id = charge_id


def refund_charge(charge_id: str, amount: int) -> RefundReport:
    return RefundReport(charge_id)
